Assets/Scripts/Controllers/CharacterSpriteController.cs

```diff
@@ -95,47 +95,46 @@ protected override void OnCreated(Character c)
 
         // Register our callback so that our GameObject gets updated whenever
         // the object's into changes.
-        c.OnCharacterChanged += OnChanged;        
+        character.OnCharacterChanged += OnChanged;
     }
 
-    protected override void OnChanged(Character c)
+    protected override void OnChanged(Character character)
     {
         // Make sure the furniture's graphics are correct.
-        SpriteRenderer inv_sr = objectGameObjectMap[c].transform.GetChild(0).gameObject.GetComponent<SpriteRenderer>();
+        SpriteRenderer inv_sr = objectGameObjectMap[character].transform.GetChild(0).gameObject.GetComponent<SpriteRenderer>();
 
         // Important to set the characters SortOrder first.
-        int charSortOrder = c.animation.SetAndGetSortOrder();
-        if (c.inventory != null)
+        int charSortOrder = character.animation.SetAndGetSortOrder();
+        if (character.inventory != null)
         {
-            inv_sr.sprite = SpriteManager.GetSprite("Inventory", c.inventory.GetName());
+            inv_sr.sprite = SpriteManager.GetSprite("Inventory", character.inventory.GetName());
             inv_sr.sortingOrder = charSortOrder + 1;
         }
         else
         {
             inv_sr.sprite = null;
         }
 
-        if (objectGameObjectMap.ContainsKey(c) == false)
+        if (objectGameObjectMap.ContainsKey(character) == false)
         {
             Debug.ULogErrorChannel("CharacterSpriteController", "OnCharacterChanged -- trying to change visuals for character not in our map.");
             return;
         }
 
-        GameObject char_go = objectGameObjectMap[c];
+        GameObject char_go = objectGameObjectMap[character];
 
-        char_go.transform.position = new Vector3(c.X, c.Y, 0);
+        char_go.transform.position = new Vector3(character.X, character.Y, 0);
     }
 
-    protected override void OnRemoved(Character c)
+    protected override void OnRemoved(Character character)
     {
-        c.OnCharacterChanged -= OnChanged;  
```

Comment:
Why removing this? How are character.OnCharacterChanged callbacks getting cleaned up?